core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -269,7 +323,11 @@ public Snapshot apply() {
         operation(),
         summary(base),
         base.currentSchemaId(),
-        manifestList.location());
+        manifestList.location(),
+        manifestListSize,
+        manifestListKeyMetadata,
+        wrappedManifestListKeyMetadata,
+        wrappedKeyEncryptionKey);
```

Comment:
I'd prefer to create a `ManifestListFile` and pass it here instead of adding so many parameters.